### PR TITLE
Preact: change the Context and useContext typing to support any type

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: src/preact/index.js [hunk 1]
```diff
@@ -110,8 +111,9 @@ export function useLayoutEffect(effect, opt_deps) {
 }
 
 /**
- * @param {PreactDef.Context} context
- * @return {!JsonObject}
+ * @param {PreactDef.Context<T>} context
+ * @return {T|undefined}
```
Comment (jridgewell) [inline src/preact/index.js:115]: Doesn't this return the default value used when creating the context? Meaning, we should never get `undefined`, unless you used `createContext(undefined)` (which I think is an error?).

Comment (dvoytenko) [reply src/preact/index.js]: Yeah. I think you're right. If the context allows default value, it should be reflected at the context creation time, not at the use time.

File: src/preact/index.js [hunk 2]
```diff
@@ -65,8 +65,9 @@ export function createRef() {
 }
 
 /**
- * @param {!Object} value
- * @return {!PreactDef.Context}
+ * @param {T=} value
```
Comment (caroqliu) [inline src/preact/index.js:68]: Is there a reason this was required before?

Comment (dvoytenko) [reply src/preact/index.js:68]: Oops, missed this one too.

**Pull request merged**